kaggle_environments/envs/hungry_geese/hungry_geese.py

```diff
@@ -183,22 +187,23 @@ def interpreter(state, env):
         agent_count = len(state)
         heads = sample(range(columns * rows), agent_count)
         shared_observation["geese"] = [[head] for head in heads]
-        shared_observation["food"] = sample(set(range(columns * rows)).difference(heads), min_food)
+        food_candidates = set(range(columns * rows)).difference(heads)
+        # Ensure we only place as many food as there are open squares
+        min_food = min(min_food, len(food_candidates))
+        shared_observation["food"] = sample(food_candidates, min_food)
         return state
 
     geese = shared_observation.geese
     food = shared_observation.food
 
-    # Update active agent rewards.
-    for index, agent in enumerate(state):
-        if agent.status == "ACTIVE":
-            agent.reward = len(env.steps) + len(geese[index])
-
     # If there is no last state, reuse current state so that current action is never the opposite of the last action.
     last_state = env.steps[-1] if len(env.steps) > 1 else state
     # Apply the actions from active agents.
     for index, agent in enumerate(state):
         if agent.status != "ACTIVE":
+            if agent.status != "INACTIVE" and agent.status != "DONE":
```

Comment:
nit: could do `not in ["INACTIVE", "DONE"]`, or do the explicit check like the comment mentions